src/sparse_ir/spr.py

```diff
@@ -1,66 +1,135 @@
-        y_sampling_points = basis.beta*omega_sampling_points/basis.wmax
+        self._poles = basis.default_omega_sampling_points() if sampling_points is None\
+            else np.asarray(sampling_points)
+        self._y_sampling_points = basis.beta*self._poles/basis.wmax
+
+        self.u = TauPoleBasis(basis.beta, basis.statistics, self._poles)
+        self.uhat = MatsubaraPoleBasis(basis.beta, self._poles)
 
         # Fitting matrix from IR
-        weight = basis.kernel.weight_func(basis.statistics)(y_sampling_points)
+        weight = basis.kernel.weight_func(self.statistics)(self._y_sampling_points)
         fit_mat = np.einsum(
             'l,lp,p->lp',
             -basis.s,
-            basis.v(omega_sampling_points),
+            basis.v(self._poles),
             weight,
             optimize=True
         )
         self.matrix = DecomposedMatrix(fit_mat)
 
+    @property
+    def statistics(self):
+        return self.basis.statistics
+
     @property
     def sampling_points(self):
         return self._poles
 
+    @property
+    def size(self):
+        """Number of basis functions / singular values."""
+        return self._poles.size
+
+    @property
+    def basis(self) -> FiniteTempBasis:
+        """ Underlying basis """
+        return self._basis
+
+    @property
+    def beta(self) -> float:
+        """Inverse temperature (this is `None` because unscaled basis)"""
+        return self.basis.beta
+
+    @property
+    def wmax(self) -> float:
+        """Frequency cutoff (this is `None` because unscaled basis)"""
+        return self.basis.wmax
+
     def from_IR(self, gl: np.ndarray, axis=0) -> np.ndarray:
         """
         From IR to SPR
 
         gl:
             Expansion coefficients in IR
-            It is assumed that KFFlat is used for both of fermion and boson.
         """
         return self.matrix.lstsq(gl, axis)
 
-    def evaluate_matsubara(
-            self,
-            g_spr: np.ndarray,
-            vsample: np.ndarray,
-            axis=0) -> np.ndarray:
+    def to_IR(self, g_spr: np.ndarray, axis=0) -> np.ndarray:
         """
-        Evaluate on Matsubara frequencies
-
-        vsample:
-            Imaginary frequencies
+        From SPR to IR
 
         g_spr:
-            SPR coefficients
+            Expansion coefficients in SPR
         """
-        coeffs = np.moveaxis(g_spr, source=axis, destination=0)
-        iv = 1j*vsample * np.pi/self._basis.beta
-        giv = np.einsum(
-            'wp, p...->w...', 1/(iv[:, None] - self._poles[None, :]), coeffs)
-        giv = np.moveaxis(giv, source=0, destination=axis)
-        return giv
+        y = self._basis.beta * self.sampling_points/self._basis.wmax
+        return np.einsum('l,p,lp,p...->l...',
+            -self._basis.s,
+            self._basis.kernel.weight_func(self.statistics)(y),
+            self._basis.v(self.sampling_points),
+            g_spr
+        )
```

Comment:
I think you may have lost support for the `axis` argument there ...

Reply:
OK, I fixed this (by using DecomposedMatrix)